torchprime/torch_xla_models/experimental/custom_kernel.py

```diff
@@ -0,0 +1,429 @@
+import dataclasses
+import functools
+import json
+from dataclasses import asdict
+from typing import Any
+
+import torch
+import torch_xla.debug.profiler as xp
+from torch.library import custom_op
+from torch.utils._pytree import tree_flatten
+from torch_xla.core.xla_builder import jax_func_to_xla_computation
+from torch_xla.distributed.spmd import Mesh
+from torch_xla.experimental.custom_kernel import requires_jax
+
+# use the global var to cache XlaComputation
+func_arg_to_XlaComputation: dict[tuple[Any], str] = {}
```

Comment:
nit: the naming convention for a private global variable is `_FUNC_ARG_TO_XLA_COMPUTATION` i think